gips/data/smap/smap.py

```diff
@@ -0,0 +1,221 @@
+
+    # NASA assets require special authentication
+    _manager_url = "https://urs.earthdata.nasa.gov"
+    _tile_attribute = 'tileid'
+
+class smapAsset(Asset):
+    Repository = smapRepository
+    _sensors = {'RAD': {'description': 'Soil Moisture Active Passive Radiometer'}}
+    _assets = {
+        'SM_P_E': {
+            'url': 'https://n5eil01u.ecs.nsidc.org/SMAP/SPL3SMP_E.002',
+            'pattern': r'SMAP\_.{2}\_SM\_P\_E\_.{8}\_.{6}\_.{3}\.h5',
+            'description': 'Passive Enhanced Radiometer Based SM at 9KM',
+            'startdate': datetime.date(2015, 3, 31),
+            'latency': 1,
+        },
+        'SM_P': {
+            'url': 'https://n5eil01u.ecs.nsidc.org/SMAP/SPL3SMP.005',
+            'pattern': r'SMAP\_.{2}\_SM\_P\_.{8}\_.{6}\_.{3}\.h5',
+            'description': 'Passive Original Radiometer Based SM at 36KM',
+            'startdate': datetime.date(2015, 3, 31),
+            'latency': 1,
+        },
+    }
+
+    def __init__(self, filename):
+        """ Inspect a single file and get some metadata """
+        super(smapAsset, self).__init__(filename)
+
+        bname = os.path.basename(filename)
+        self.asset = (re.search('(?<=SMAP_L3_)\w*(?=_[0-9]{8})', bname)).group(0)
+        date_here = (re.search('[0-9]{8}', bname)).group(0)
+        self.date = datetime.datetime.strptime(date_here, "%Y%m%d").date()
+        self._version = (re.search('R[0-9]*', bname)).group(0)
+        self.tile = 'h01v01'
+
+    @classmethod
+    def query_provider(cls, asset, tile, date):
+        """Find out from the SMAP servers what assets are available.
+
+        Uses the given (asset, date) tuple as a search key, andcat
+        returns a tuple:  base-filename, url
+        """
+
+        mainurl = "%s/%s" % (cls._assets[asset]['url'], str(date.strftime('%Y.%m.%d')))
+
+        pattern = r'SMAP\_.{2}\_%s\_%s\_.{6}\_.{3}\.h5' % (asset, str(date.strftime('%Y%m%d')))
+        cpattern = re.compile(pattern)
+        err_msg = "Error downloading: " + mainurl
+        with utils.error_handler(err_msg):
+            response = cls.Repository.managed_request(mainurl, verbosity=2)
+            if response is None:
+                return None, None
+
+        for item in response.readlines():
+            # screen-scrape the content of the page and extract the full name of the needed file
+            # (this step is needed because part of the filename, the creation timestamp, is
+            # effectively random).
+            if cpattern.search(item):
+                if 'xml' in item:
+                    continue
+                basename = cpattern.findall(item)[0]
+                url = ''.join([mainurl, '/', basename])
+                return basename, url
+        utils.verbose_out('Unable to find remote match for '
+                          '{} at {}'.format(pattern, mainurl), 4)
+        return None, None
+
+
+    @classmethod
+    def fetch(cls, asset, tile, date):
```

Comment:
How come fetch was overwritten?  Presuming it was necessary, add a docstring explaining the need.